doc/espnet2_training_option.md

```diff
@@ -0,0 +1,209 @@
+
+Note that ESPnet2 always selects`_` instead of `-` for the separation for the name of option to avoid confusing.
+
+A notable feature of `--print_config` is that it shows the configuration parsing with the given arguments dynamically: You can look up the parameters for changeable class for example.
+
+```bash
+# Show parameters of Adam optimizer
+python -m espnet2.bin.asr_train --optim adam --print_config
+# Show parameters of ReduceLROnPlateau scheduler
+python -m espnet2.bin.asr_train --scheduler ReduceLROnPlateau --print_config
+```
+
+## Configuration file
+You can find the configuration files for DNN training in `conf/train_*.yaml`.
+
+```bash
+ls conf/
+```
+
+We adopts [ConfigArgParse](https://github.com/bw2/ConfigArgParse) for this configuration system. The configuration in yaml format has equivalent effect to the command line argument. e.g. The following two are equivalent:
+
+```yaml
+# config.yaml
+foo: 3
+bar: 4
+```
+
+```bash
+python -m espnet2.bin.asr_train --config conf/config.yaml
+python -m espnet2.bin.asr_train --foo 3 --bar 4
+```
+
+
+## Change the configuration for dict type value
+
+Come parameters are named as `*_conf`, e.g. `optim_conf`, `decoder_conf` and they has the `dict` type value. We also provide the way to configure the nested value in such dict object.
+
+```bash
+# e.g. Change parameters one by one
+python -m espnet2.bin.asr_train --optim_conf lr=0.1 --optim_conf rho=0.8
+# e.g. Give the parameters in yaml format
+python -m espnet2.bin.asr_train --optim_conf "{lr: 0.1, rho: 0.8}"
+```
+
+## Resume training process
+
+```bash
+python -m espnet2.bin.asr_train --resume true
+```
+
+The state of training process is saved at the end of every epoch as `checkpoint.pth` and the training process can be resumed from the start of the next epoch. Checkpoint includes the following states.
+
+- Model state
+- Optimizer states
+- Scheduler states
+- Reporter state
+- apex.amp state
+
+## Change logging interval
+The result in the middle state of the training will be shown by the specified number:
+
+```bash
+python -m espnet2.bin.asr_train --log_interval 100
+```
+
+## Change the number of iterations in each epoch
+
+By default, an ``epoch`` indicates using up whole data in the training corpus and the following steps will also run after training for every epochs:
+
+- Validation
+- Saving model and checkpoint
+- Show result in the epoch
+
+Sometimes the examination after training with whole corpus is too coarse if using large corpus: `--num_iters_per_epoch` restrict the number of iteration of each epochs.
+
+```bash
+python -m espnet2.bin.asr_train --num_iters_per_epoch 1000
+```
+
+Note that he training process can't be resumed at the middle of an epoch because data iterators is stateless, but instead of it, the iterators for each epochs can be built with the specific epoch number deterministically, just like:
```

Comment:
```suggestion
Note that the training process can't be resumed at the middle of an epoch because data iterators are stateless, but instead of it, the iterators for each epoch can be built with the specific epoch number deterministically, just like:
```